Specs/Core/GoogleEarthEnterpriseTerrainProviderSpec.js

```diff
@@ -0,0 +1,318 @@
+            expect(terrainProvider.credit).toBeDefined();
+        });
+    });
+
+    it('has a water mask is false', function() {
+        installMockGetQuadTreePacket();
+
+        terrainProvider = new GoogleEarthEnterpriseTerrainProvider({
+            url : 'made/up/url'
+        });
+
+        return pollToPromise(function() {
+            return terrainProvider.ready;
+        }).then(function() {
+            expect(terrainProvider.hasWaterMask).toBe(false);
+        });
+    });
+
+    it('has vertex normals is false', function() {
+        installMockGetQuadTreePacket();
+
+        terrainProvider = new GoogleEarthEnterpriseTerrainProvider({
+            url : 'made/up/url'
+        });
+
+        return pollToPromise(function() {
+            return terrainProvider.ready;
+        }).then(function() {
+            expect(terrainProvider.hasVertexNormals).toBe(false);
+        });
+    });
+
+    describe('requestTileGeometry', function() {
+        it('uses the proxy if one is supplied', function() {
+            installMockGetQuadTreePacket();
+            var baseUrl = 'made/up/url';
+
+            loadWithXhr.load = function(url, responseType, method, data, headers, deferred, overrideMimeType) {
+                expect(url.indexOf('/proxy/?')).toBe(0);
+
+                loadWithXhr.defaultLoad('Data/GoogleEarthEnterprise/gee.terrain', responseType, method, data, headers, deferred);
+            };
+
+            terrainProvider = new GoogleEarthEnterpriseTerrainProvider({
+                url : baseUrl,
+                proxy : new DefaultProxy('/proxy/')
+            });
+
+            return pollToPromise(function() {
+                return terrainProvider.ready;
+            }).then(function() {
+                return terrainProvider.requestTileGeometry(0, 0, 0);
+            });
+        });
+
+        it('provides GoogleEarthEnterpriseTerrainData', function() {
+            installMockGetQuadTreePacket();
+            loadWithXhr.load = function(url, responseType, method, data, headers, deferred, overrideMimeType) {
+                loadWithXhr.defaultLoad('Data/GoogleEarthEnterprise/gee.terrain', responseType, method, data, headers, deferred);
+            };
+
+            return waitForTile(0, 0, 0, function(loadedData) {
+                expect(loadedData).toBeInstanceOf(GoogleEarthEnterpriseTerrainData);
+            });
+        });
+
+        it('returns undefined if too many requests are already in progress', function() {
+            installMockGetQuadTreePacket();
+            var baseUrl = 'made/up/url';
+
+            var deferreds = [];
+            var loadRealTile = true;
+            loadWithXhr.load = function(url, responseType, method, data, headers, deferred, overrideMimeType) {
+                if (loadRealTile) {
+                    loadRealTile = false;
+                    return loadWithXhr.defaultLoad('Data/GoogleEarthEnterprise/gee.terrain', responseType, method, data, headers, deferred);
```

Comment:
I got it from the public test server that we reference in the Sandcastle example, but I actually have code to generate a tile in this test file, so I can just do that.